lib/src/abstract_scope_manager.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:opentracing/opentracing.dart';
+
+/// The [ScopeManager] abstracts both the activation of [Span] instances via
+/// [ScopeManager].activate(Span, boolean)} and access to an active
+/// [Span]/[Scope] via [ScopeManager].active.
+abstract class ScopeManager {
+  /// Make a Span instance active.
+  ///
+  /// [span] that should become the [active] [Span]
+  /// When [finishSpanOnClose] is true, the [span] will automatically be
+  /// finished when [Scope].close() is called.
+  /// Returns a [Scope] instance to control the end of the active period for the
+  /// [Span]. It is a programming error to neglect calling [Scope].close() on
+  /// the returned instance.
+  Scope activate(Span span, {bool finishSpanOnClose});
```

Comment:
I noticed that the spec lists finishSpanOnClose [as required](https://github.com/opentracing/specification/blob/master/rfc/scope_manager.md#scopemanager).
> A **required** boolean parameter finish span on close will mark whether...

This should be made a positional parameter or have the `@required` annotation added.

Reply:
I agree with your suggestion, but later came across this:  https://github.com/opentracing/opentracing-java/issues/291

I also noticed that your reference isn't the spec, but rather a suggested change to it.  I'm inclined to remove this parameter for now, until that spec suggestion is accepted and/or we have an immediate use-case for it.  Thoughts?

Reply:
After creating an actual implementation in basictracer_dart, I've changed my position on your suggestion and think we should proceed with it.